integration-cli/docker_cli_build_test.go

```diff
@@ -5660,3 +5660,434 @@ func (s *DockerSuite) TestBuildNullStringInAddCopyVolume(c *check.C) {
 	_, err = buildImageFromContext(name, ctx, true)
 	c.Assert(err, check.IsNil)
 }
+
+func (s *DockerSuite) TestBuildBuildTimeArg(c *check.C) {
+	imgName := "bldargtest"
+	envKey := "foo"
+	envVal := "bar"
+	args := []string{
+		"--build-arg", fmt.Sprintf("%s=%s", envKey, envVal),
+	}
+	dockerfile := fmt.Sprintf(`FROM busybox
+		ARG %s=
```

Comment:
@tiborvass not sure we can say for sure what will happen in the future. Any of the stuff you described might be possible, but I'm not sure we'd want encode that much "magic" into the presence/absence of the "=". I'd like to avoid some of the headache we've seen with the magic of ADD/COPY.

I'd much prefer to be explicit about it by adding some kind of "required" flag to the `ARG` command and keep the obvious things obvious.  Meaning `ARG foo` defines "foo".  `ARG foo=bar` defines foo and gives it a default value. And then we can figure out the right syntax to say "ARG MUST have a value" later once we have a solid use case for it.